### PR TITLE
Multipool config & bazel 0.16 update

### PR DESCRIPTION
**Pull request by billfeng327**
Added multipool config for large instances and standard instances, replacing the sentinel file for large instances. Also updated the RBE container to the image containing bazel 0.16.1

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: test/core/gpr/BUILD [hunk 1]
```diff
@@ -81,12 +81,12 @@ grpc_cc_test(
 grpc_cc_test(
     name = "mpscq_test",
     srcs = ["mpscq_test.cc"],
+    data = ["//third_party/toolchains/machine-size:large"],
```
Comment (bwkimmel) [inline test/core/gpr/BUILD]: This should be `exec_compatible_with = ["<constraint>"],`.

I'm not how `grpc_cc_test` relates to `cc_test` -- you may need to add an `exec_compatible_with` to its definition to pass it along to an underlying `cc_test`.

Same for other build targets below.

**Comment by bwkimmel** [comment]
Minor nit: PR title should be multipool.

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,964,615      Total (=)      1,964,615

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,886,807      Total (>)     10,886,802

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

File: test/cpp/qps/BUILD [hunk 2]
```diff
@@ -168,10 +168,10 @@ grpc_cc_test(
 grpc_cc_test(
     name = "qps_openloop_test",
     srcs = ["qps_openloop_test.cc"],
-    data = ["//third_party/toolchains:RBE_USE_MACHINE_TYPE_LARGE"],
+    exec_compatible_with = ["//third_party/toolchains/machine-size:large"],
     deps = [
         ":benchmark_config",
-        ":driver_impl",
+        ":driver_impl",r
```
Comment (bwkimmel) [inline test/cpp/qps/BUILD]: Typo.

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,964,588      Total (=)      1,964,588

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,886,809      Total (>)     10,886,805

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,964,580      Total (=)      1,964,580

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,886,957      Total (>)     10,886,949

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,964,600      Total (=)      1,964,600

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,886,957      Total (>)     10,886,954

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by jtattermusch** [comment]
I don't see anything suspicious in the diff, but perhaps I'm lacking context. What's the end goal of this change?

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,965,918      Total (=)      1,965,918

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,891,793      Total (<)     10,891,795

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,965,918      Total (=)      1,965,918

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,891,795      Total (>)     10,891,791

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,965,918      Total (=)      1,965,918

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,891,798      Total (<)     10,891,801

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: tools/dockerfile/test/bazel/Dockerfile [hunk 3]
```diff
@@ -46,7 +46,7 @@ RUN pip install futures==2.2.0 enum34==1.0.4 protobuf==3.5.2.post1 six==1.10.0 t
 # Bazel installation
 RUN echo "deb [arch=amd64] http://storage.googleapis.com/bazel-apt stable jdk1.8" > /etc/apt/sources.list.d/bazel.list
 RUN curl https://bazel.build/bazel-release.pub.gpg | apt-key add -
-RUN apt-get -y update && apt-get -y install bazel=0.15.0 && apt-get clean
+RUN apt-get -y update && apt-get -y install bazel=0.16.1 && apt-get clean
```
Comment (jtattermusch) [inline tools/dockerfile/test/bazel/Dockerfile]: FYI (as discussed offline), this Dockerfile is only used for the jobs `grpc_bazel_build` and `grpc_python_bazel_test` to build with bazel locally (and run tests python tests with bazel locally too). It doesn't influence the Foundry builds at all.

**Comment by billfeng327** [comment]
Replaced by PR [17110](https://github.com/grpc/grpc/pull/17110)

**Pull request closed without merging**